ee/clickhouse/queries/funnels/funnel_correlation.py

```diff
@@ -521,19 +539,126 @@ def _run(self) -> Tuple[List[EventOddsRatio], bool]:
+            # Ensure we have exhausted all values of self._filter.correlation_type
+            _assert_never(self._filter.correlation_type)
+
+    def construct_event_correlation_people_url(self, success: bool, event_definition: EventDefinition) -> str:
+        # NOTE: we need to convert certain params to strings. I don't think this
+        # class should need to know these details, but with_data is
+        # expecting the values as they are serialized in the url
+        # TODO: remove url serialization details from this class, it likely
+        # belongs in the application layer, or perhaps `FunnelCorrelationPeople`
+        params = self._filter.with_data(
+            {
+                "funnel_correlation_person_converted": "true" if success else "false",
+                "funnel_correlation_person_entity": json.dumps({"id": event_definition["event"], "type": "events"}),
+            }
+        ).to_params()
+        return f"/api/person/funnel/correlation/?{urllib.parse.urlencode(params)}"
+
+    def construct_event_with_properties_people_url(self, success: bool, event_definition: EventDefinition) -> str:
+        if self.support_autocapture_elements():
+            # If we have an $autocapture event, we need to special case the
+            # url by converting the `elements` chain into an `Action`
+            event_name, _, _ = event_definition["event"].split("::")
+            elements = event_definition["elements"]
+            first_element = elements[0]
+            elements_as_action = {
+                "tag_name": first_element["tag_name"],
+                "href": first_element["href"],
+                "text": first_element["text"],
+                "selector": build_selector(elements),
+            }
+            params = self._filter.with_data(
                 {
-                    "success_count": odds_ratio["success_count"],
-                    "failure_count": odds_ratio["failure_count"],
-                    "odds_ratio": odds_ratio["odds_ratio"],
-                    "correlation_type": odds_ratio["correlation_type"],
-                    "event": self.serialize_event_with_property(odds_ratio["event"]),
+                    "funnel_correlation_person_converted": "true" if success else "false",
+                    "funnel_correlation_person_entity": json.dumps(
+                        {
+                            "id": event_name,
+                            "type": "events",
+                            "properties": [
+                                {
+                                    "key": property_key,
+                                    "value": [property_value],
+                                    "type": "element",
+                                    "operator": "exact",
+                                }
+                                for property_key, property_value in elements_as_action.items()
+                                if property_value
+                                if not None
```

Comment:
what does this mean? O.o

Reply:
haha fixing... Was meant to be property_value is not None